### PR TITLE
Copy resource files when transforming byte code.

### LINKED ISSUE #2356: Kotlin: Package level functions cannot be imported from library project with new gradle plugin
This is a pretty elaborate case so hopefully I can explain it well enough.

We have a project consisting of a main application and a library project.  The library project contains Realm models that are shared.  Using the new plugin on both the main application and the library project causes the main project to not see the package level functions in the library.

I've attached a small sample project to show what I mean.

Given this file in the library project 

~~~kotlin
package io.vokal.realmtest.model

import io.realm.*
import io.realm.annotations.*

@RealmClass
open class Dog(
    @PrimaryKey
    public open var name: String = "spot"
): RealmObject() { }

@RealmModule(library = true, allClasses = true)
public class LibraryModule {
}

public fun Realm.findSpot(): Dog {
    return where(Dog::class.java).equalTo("name", "spot").findFirst()
}
~~~

I should be able to import `findSpot` in the main activity.  

Instead I receive `e: /home/wmbest2/realm-library-test/app/src/main/kotlin/io/vokal/realmtest/MainActivity.kt: (37, 23): Unresolved reference: findSpot`

[realm-library-test.zip](https://github.com/realm/realm-java/files/149012/realm-library-test.zip)

### PR DESCRIPTION
**Pull request by zaki50**
Realm Transformer must copy non-class files to output dir since Kotlin compiler generates `*.kotlin_module` file in `META-INF`.

This fixes #2356

### CODE REVIEW

**Comment by cmelchior** [comment]
I am very surprised we need to do this and it smells like a bug in either Kotlin or the Transform API, but apart from that :+1:

File: realm-transformer/src/main/groovy/io/realm/transformer/RealmTransformer.groovy [hunk 1]
```diff
@@ -197,4 +199,26 @@ class RealmTransformer extends Transform {
...
+    private copyResourceFiles(Collection<TransformInput> inputs, TransformOutputProvider outputProvider) {
+        inputs.each {
+            it.directoryInputs.each {
+                def dirPath = it.file.absolutePath
+                it.file.eachFileRecurse(FileType.FILES) {
+                    if (!it.absolutePath.endsWith(SdkConstants.DOT_CLASS)) {
+                        logger.info "  Copying resource ${it}"
+                        def dest = new File(getOutputFile(outputProvider),
+                                it.absolutePath.substring(dirPath.length()))
+                        FileUtils.copyFile(it, dest);
```
Comment (emanuelez) [inline realm-transformer/src/main/groovy/io/realm/transformer/RealmTransformer.groovy]: I wonder which library includes `FileUtils`. I generally prefer to use Guava's `Files`, but I don't have a strong opinion about it.

**Comment by emanuelez** [comment]
This is quite confusing. We specify the scope to be only classes so how do other resource appear in the inputs?

**Comment by kneth** [comment]
Please add a line in `CHANGELOG.md` as it closes an issue.

**Comment by cmelchior** [comment]
This is still unreleased code, so IMO it doesn't need an changelog (it was a bug found in our SNAPSHOT).

**Comment by beeender** [comment]
:+1:

**Comment by emanuelez** [comment]
:+1:

**Comment by kneth** [comment]
@cmelchior Hmm, maybe an acknowledgement to "snapshot testers"?

**Comment by cmelchior** [comment]
I don't have any strong opinions either way on that.

**Comment by cmelchior** [comment]
@zaki50, @kneth will look at the above outside this PR, so I think this is safe to merge?

File: realm-transformer/src/main/groovy/io/realm/transformer/RealmTransformer.groovy (hunk 1 above)
Comment (zaki50) [inline realm-transformer/src/main/groovy/io/realm/transformer/RealmTransformer.groovy]: done

**Comment by zaki50** [comment]
I tested this issue with attached project. That project includes app project which uses
1. library project
2. library project in library project
3. binary aar

[realm-library-test_issue2356.zip](https://github.com/realm/realm-java/files/165368/realm-library-test_issue2356.zip)

**Comment by cmelchior** [comment]
@zaki50 I just finished verifying the project. Looks :+1:

**Comment by cmelchior** [comment]
:+1

**Pull request merged**